src/components/molecules/Visualizer/Engine/Cesium/hooks.ts

```diff
@@ -51,24 +51,42 @@ export default ({
 
   // imagery layers
   const [imageryLayers, setImageryLayers] =
-    useState<[string, ImageryProvider, number | undefined, number | undefined][]>();
+    useState<
+      [string, ImageryProvider, number | undefined, number | undefined, number | undefined][]
```

Comment:
It's hard to understand the slice type when you get to this point. You can define a type:

```suggestion
      ImageryLayerData[]
```

```ts
type ImageryLayerData = {
  id: string;
  provider: ImageryProvider;
  min: number | undefined;
  max: number | undefined;
  opacity: number  | undefined;
}
```